cmd/proxy/tcp_proxy.go

```diff
@@ -48,7 +48,8 @@ func (proxy *TCPProxy) clientLoop(client *net.TCPConn, quit chan bool) {
 				from.CloseWrite()
 			}
```

Comment:
This change looks good. Given that the half-close from either side is handled with the change do we still need this if block ? Also, `from.CloseWrite()` doesn't look correct here.